Include documentation for Android NDK Crash reporting (#24036)
DataDog/documentation: content/en/real_user_monitoring/error_tracking/mobile/android.md

1. Add the latest version of the [RUM Android SDK][4] to your Gradle dependencies.
2. Configure your application's `env` and `variant` when [initializing the SDK][5].
3. Run the Gradle tasks to upload your Proguard/R8 mapping file and NDK symbol files to Datadog to access deobfuscated stack traces. 

For any given error, you can access the file path, line number, and a code snippet for each frame of the related stack trace.

### Add NDK crash reporting

> Your Android application may be running native code (C/C++) for performance or code reusability reasons. In order to enable NDK crash reporting, use the Datadog NDK plugin. 

Reviewer: ```suggestion
Your Android application may be running native code (C/C++) for performance or code reusability reasons. To enable NDK crash reporting, use the Datadog NDK plugin. 
```

> 1. Add the Gradle dependency by declaring the library as a dependency in your `build.gradle` file:

Reviewer: ```suggestion
1. Declare the library as a dependency in your `build.gradle` file:
```

1. Add the latest version of the [RUM Android SDK][4] to your Gradle dependencies.
2. Configure your application's `env` and `variant` when [initializing the SDK][5].
3. Run the Gradle tasks to upload your Proguard/R8 mapping file and NDK symbol files to Datadog to access deobfuscated stack traces. 

For any given error, you can access the file path, line number, and a code snippet for each frame of the related stack trace.

### Add NDK crash reporting

> Your Android application may be running native code (C/C++) for performance or code reusability reasons. In order to get NDK crash reporting leverage the Datadog NDK plugin. 

Reviewer: ```suggestion
Your Android application may be running native code (C/C++) for performance or code reusability reasons. In order to enable NDK crash reporting, use the Datadog NDK plugin. 
```

> 1. Declare `dd-sdk-android-ndk` as a dependency in your application module's build.gradle file.

Reviewer 2: minor: in fact it can be in any module, requirement for the application module is enforced only for the Gradle plugin.

> 1. Declare `dd-sdk-android-ndk` as a dependency in your application module's build.gradle file.

Reviewer: ```suggestion
1. Declare `dd-sdk-android-ndk` as a dependency in your application module's `build.gradle` file.
```

   ```groovy
    dependencies {
        implementation "com.datadoghq:dd-sdk-android-ndk:x.x.x" 
        //(...)
    }
   ```

> 2. Enable NDK Crash collection after you have initialized the RUM SDK.

Reviewer: ```suggestion
2. Enable NDK crash collection after initializing the RUM SDK.
```

> 2. Enable NDK Crash collection after you have initialized the RUM SDK.
> 
>     ``` kotlin
>     NdkCrashReports.enable()
>     ```
> 
> 
> 
> ### Add ANR reporting

Reviewer 2: ```suggestion
2. Enable NDK Crash collection after you have initialized the RUM SDK.

    ``` kotlin
    NdkCrashReports.enable()
    ```

### Add ANR reporting
```

   ./gradlew uploadMappingRelease
   ```

> 5. If running native code, run the NDK symbol upload task.

Reviewer: ```suggestion
5. If running native code, run the NDK symbol upload task:
```

   ```bash
   ./gradlew uploadMappingRelease
   ```
   

> 5. If running native code, run the NDK symbol upload task.

Reviewer: ```suggestion
5. If running native code, run the NDK symbol upload task:
```
